src/nodes/Script.hack

```diff
@@ -86,4 +86,54 @@ final class Script extends ScriptGeneratedBase {
       },
     );
   }
+
+  /**
+   * Find all nodes that have generics and return both the nodes and their
+   * generics.
+   */
+  <<__Memoize>>
+  public function getNodesWithGenerics(): vec<(Node, TypeParameters)> {
+    $root = $this->getDeclarationsx();
+
+    // Sadly these don't have a consistent API so we need a separate Vec\map
+    // for each node type.
```

Comment:
Could be worth adding some - e.g. for functions/methods, IFunctionishDeclaration could be extended - MethodishDeclaration already has manual overrides with a generated base class, FunctionDeclaration would need the same

Reply:
Yeah I tried that first but it turned out to be a lot of changes and this was much simpler. We can always refactor this later.